### DISCUSSION TITLE
Introductions

### DISCUSSION THREAD

**Discussion by ehmatthes** [ehmatthes/django-simple-deploy#130] [General]
This discussion is from DjangoCon 2022. The current sprint discussion is in #432.

---

Hello! Thank you for your interest in `django-simple-deploy`. If you are interested in contributing to the project, please introduce yourself here. This project touches a lot of different areas, including but not limited to:
- Python code
- shell code
- All three major operating systems: Windows, macOS, and Linux
- Multiple platforms, currently including Fly.io, Platform.sh
- Interesting decisions around the boundaries of responsibility: what is the Django community responsible for, and what are platforms responsible for?

So, there are many opportunities for contributing to this project. If you are interested in contributing long-term, we will definitely find a place for you in this project. (If you're looking to make a quick open source PR and then move on, this might not be the best project to do that.)

Please introduce yourself by answering some or all of these questions. You don't need to write an essay, but this is the kind of information we need to share in order to figure out how to help people make the most meaningful and satisfying contributions to the project.

- Who are you? How would you like to introduce yourself?
- What interests you most about django-simple-deploy?
- Is there a specific way you want to contribute?
- Are there any specific strengths you bring to this project?
- Have you tried using django-simple-deploy yet? How did it go?
- Do you have a platform you want to focus on?
- Do you have the ability to push test projects to this platform? (This should not accrue significant charges because resources should be created and destroyed in a matter of minutes, but you are responsible for any charges accrued for projects that you create. This is not necessary for contributing to the project.)

Thanks!

Eric

**Comment by jerivas** [comment]
Hey 👋

Ed Rivas here. I'm interested in `django-simple-deploy` as a tool to easily get sites up in different hosting providers, and as a tool for beginners to release their projects to the world. I have experience automating deployments on both [Digital Ocean App Platform](https://www.digitalocean.com/products/app-platform) and [Render](https://render.com), so I think I can contribute something related to those two.

Looking forward to helping during the DjangoCon 2022 sprints.

**Comment by ronardcaktus** [comment]
Hola mundo 👋🏽, 

My name is Ronard Luna and I am a developer. When it comes to deployment, particularly when someone is new, there is this seemingly invisible yet almost tangible barrier to deploying a project - I love that this project aims at lowering the bar just enough so no matter how new someone is, they can think of the end goal as achievable. Moreover, this project can also normalize deployment for more experienced developers, since it deploys to an array of platforms. 

I am knowledgeable in Python and Django. Always looking forward to meeting new people 😁.

**Comment by tataraba** [comment]
Who are you? How would you like to introduce yourself?

> My name is Mario Munoz and I created the website [Python By Night](https://www.pythonbynight.com). I like to write code when I have a spare moment, which is usually at night.

What interests you most about django-simple-deploy?

> As a relatively new developer, I've gone through deployments with different platforms and struggled to get through the documentation to see what needs to be done to do a "simple" deployment. I would love to understand what types of decisions and obstacles there are for newer developers who face these decisions. I am intrigued and excited by Eric's desire to reduce the friction that newer (as well as experienced) developers face when wanting to deploy to a new platform.

Is there a specific way you want to contribute?

> For the most part, I believe I can help with the conversation around these topics, as well as providing insight into some of the platforms I've used, such as Google App Engine, Digital Ocean, and Platform.sh. I can likely help with documentation, and would look for opportunities for smaller pieces with the code.

Are there any specific strengths you bring to this project?

> My strength is that I barely have any django experience, and I'm a newer Python developer. (Wait, is that a weakness?) I can use the perspective of what challenges a newer developer would have going into a deploy, and what expectations one may have.

Have you tried using django-simple-deploy yet? How did it go?

> Not yet.

Do you have a platform you want to focus on?

> I would be open to looking at Digital Ocean and Platform.sh, and maybe Azure down the road (I haven't used that one yet, but would be interested in it).

Do you have the ability to push test projects to this platform? (This should not accrue significant charges because resources should be created and destroyed in a matter of minutes, but you are responsible for any charges accrued for projects that you create. This is not necessary for contributing to the project.)

> Only platform.sh

**Comment by leetrout** [comment]
Hey friends! I'm Lee and I enjoy Python & Go tooling / automation work.

I chatted with Eric at DjangoCon and would love to help out with any of the under-the-hood bits no one else is intersted in. We specifically chatted about maybe helping with logging.

I haven't tried the tool yet but I have a small project I am working on and will try it out with Fly.io!

**Reply by ehmatthes** [reply]
Nice to see you here, @leetrout. :)

**Comment by glasnt** [comment]
> Who are you? How would you like to introduce yourself?

Hi! 👋  I'm Katie (she/they), general Djangodork and automation engineer

> What interests you most about django-simple-deploy?

There's a lot humans need to learn in order to do Django; let's not make them get a Certificate in System Administration on top of that :)

> Is there a specific way you want to contribute?

Helping test existing platforms, maybe some test stuff, but mostly new platform (see below)

> Are there any specific strengths you bring to this project?

Recovering SRE, current Developer Advocate for Google Cloud

> Have you tried using django-simple-deploy yet? How did it go?

#152 

> Do you have a platform you want to focus on?

Google Cloud Run

> Do you have the ability to push test projects to this platform? (This should not accrue significant charges because resources should be created and destroyed in a matter of minutes, but you are responsible for any charges accrued for projects that you create. This is not necessary for contributing to the project.)

Yes.

**Reply by ehmatthes** [reply]
Hi Katie, it was really enjoyable to talk with you about this project at DjangoCon, and your run-throughs have been so helpful already. I look forward to working with on this. :)
